compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,42 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        @tailrec
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)],
+            namesCount: Map[String, Int] = Map.empty): Unit = {
+          pairs match {
+            case (js.StringLiteral(keyName), p) :: tail =>
+              val oldValue = namesCount.getOrElse(keyName, 0)
+              val newCount = namesCount.updated(keyName, oldValue + 1)
+              warnIfDuplicatedKey(tail, newCount)
```

Comment:
Can we collect all of this using a single composition of functional combinators? Like
```scala
val keyCounts = pairs.map(_._1).groupBy(_.value).mapValues(_.size)
```
or, for a more predictable order of the keys:
```scala
val allKeys = pairs.map(_._1)
val keyCounts = allKeys.distinct.map(key => key -> allKeys.count(_ == key))
```